src/context/polling.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useContext, useEffect, useState } from "react"
+import { useAdminBatchJobs } from "medusa-react"
+import { AdminGetBatchParams } from "@medusajs/medusa"
+import { BatchJob } from "@medusajs/medusa/dist"
+import { AccountContext } from "./account"
+
+export const defaultPollingContext: {
+  batchJobs?: BatchJob[]
+  hasPollingError?: boolean
+} = {}
+
+export const PollingContext = React.createContext(defaultPollingContext)
+
+export const PollingProvider = ({ children }) => {
+  const { isLoggedIn } = useContext(AccountContext)
+
+  const [shouldPollBatchJobs, setShouldPollBatchJobs] = useState(false)
+  const [polledBatchJobs, setPolledBatchJobs] = useState<BatchJob[] | undefined>()
+  const [hasPollingError, setHasPollingError] = useState<boolean | undefined>()
+
+  const oneMonthAgo = new Date(new Date().setMonth(new Date().getMonth() - 1))
+  oneMonthAgo.setHours(0, 0, 0, 0);
+
+  const {
+    batch_jobs: batchJobs,
+    isError: isBathJobPollingError,
+    isFetching
+  } = useAdminBatchJobs({
+    created_at: { gte: oneMonthAgo },
+    failed_at: "null",
```

Comment:
commect: this will be changed to null instead but require https://github.com/medusajs/medusa/pull/1748